### ISSUE TITLE
FLEX prematurely purges older items from network debugging cache

### ISSUE THREAD

**Issue by ThePantsThief** [Flipboard/FLEX#173]
FLEX purges older items from the network debugging cache even when the cache is nowhere near full

**Comment by ThePantsThief** [comment]
I would assign this to myself if I could

**Comment by NSExceptional** [comment]
I propose we give this a try as a replacement: https://github.com/nicklockwood/OSCache

Thoughts?

**Comment by aleksandergrzyb** [comment]
This is not connected to https://github.com/Flipboard/FLEX/issues/91?

**Comment by NSExceptional** [comment]
After reviewing the comments it seems to be related, yeah. What happened there? It's 2 years old

**Comment by aleksandergrzyb** [comment]
Yeah, I know. 😞 I made a fork and implemented it privately using PINCache for my company internal purposes. The difference is that you cannot opt-out from persistence and PINCache dependency. Unfortunately I didn't have time to implement this and create a PR. I don't think I will have time anywhere soon. 😞

**Comment by NSExceptional** [comment]
I can take care of it. What would it entail?

**Comment by ryanolsonk** [comment]
@NSExceptional I've been annoyed by this as well (particularly when backgrounding an app). I'm ok pulling in OSCache if you want to give that a shot.

**Comment by NSExceptional** [comment]
Awesome. I'll send a PR when I have time to do it.

**Comment by hiburn8** [comment]
This should be marked as an issue rather than enhancement. Since the user can control the amount of cache space they want to use with a slider (although this is fairly limited given some iOS devices have 4GB of RAM), and they can manually purge the cache, there really is no reason to use NSCache here. And it introduces a behaviour that the user doesn't expect.
I would argue we can just use a NSMutableDictionary here and purge it under the behaviour that the user expects.
